frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -79,22 +87,75 @@ type ConvertOpt struct {
 }
 
 func Dockerfile2LLB(ctx context.Context, dt []byte, opt ConvertOpt) (*llb.State, *Image, *binfotypes.BuildInfo, error) {
-	ds, bi, err := toDispatchState(ctx, dt, opt)
+	ds, _, bi, err := toDispatchState(ctx, dt, opt)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 	return &ds.state, &ds.image, bi, nil
 }
 
 func Dockefile2Outline(ctx context.Context, dt []byte, opt ConvertOpt) (*outline.Outline, error) {
-	ds, _, err := toDispatchState(ctx, dt, opt)
+	ds, _, _, err := toDispatchState(ctx, dt, opt)
 	if err != nil {
 		return nil, err
 	}
 	o := ds.Outline(dt)
 	return &o, nil
 }
 
+type SBOMTargets struct {
+	Core   llb.State
+	Extras map[string]llb.State
```

Comment:
Is the name in this map retrievable from the final attestation (descriptor)?

Reply:
So by default this is the last part of the path component - I think we should *probably* document this, so that scanners can use this as the name in the SBOM.